packages/nuxt/src/core/nuxt.ts

```diff
@@ -190,9 +190,10 @@ async function initNuxt (nuxt: Nuxt) {
 
   // Automatically register user modules
   for (const config of nuxt.options._layers.map(layer => layer.config).reverse()) {
+    const modulesDir = (config.rootDir === nuxt.options.rootDir ? nuxt.options : config).dir?.modules || 'modules'
```

Comment:
Note: We shall find a better way to mark top level layer (in resolver maybe) to avoid rootDir === rootDir checks

Reply:
that would be great 👍